src/uhs/transaction/validation.cpp

```diff
@@ -33,8 +33,13 @@ namespace cbdc::transaction::validation {
         return std::tie(m_code, m_idx) == std::tie(rhs.m_code, rhs.m_idx);
     }
 
-    auto check_tx(const cbdc::transaction::full_tx& tx)
+    auto check_tx(const cbdc::transaction::full_tx& tx,
+                  const std::unordered_set<pubkey_t, hashing::null>& minters)
```

Comment:
Is it possible to use a more general container here (like a vector, array etc)? Using a `std::unordered_set` with a specific custom hasher seems too specific to me.